src/fsharp/FSharp.Core/Linq.fs

```diff
@@ -195,7 +195,7 @@ module LeafExpressionConverter =
 
     let isFunctionType typ = equivHeadTypes typ (typeof<(int -> int)>)
     let getFunctionType typ = 
-        if not (isFunctionType typ) then invalidArg "typ" "cannot convert recursion except for function types"
+        if not (isFunctionType typ) then invalidArg (nameof typ) "cannot convert recursion except for function types"
```

Comment:
i dont think we should use `nameof` in this pr.
Like that you cannot compile `FSharp.Core` with LKG compiler (i think that's used somewhere in proto)
Also that make this pr bigger than needed 
The use of `nameof` can be an additional pr after this one is merged